admin/metabox/class-metabox.php

```diff
@@ -187,7 +201,7 @@ public function add_meta_box() {
 
 		if ( is_array( $post_types ) && $post_types !== array() ) {
 			foreach ( $post_types as $post_type ) {
-				if ( $this->is_metabox_hidden( $post_type ) === false ) {
+				if ( $this->should_display_metabox( $post_type ) === true ) {
```

Comment:
Please change the construction of this condition to the following:
```
if ( $this->should_display_metabox( $post_type ) === false ) {
    continue;
}
```

This will reduce complexity.

Reply:
Another, perhaps cleaner, option would be to create should_display_metabox() as a copy of is_metabox_hidden() instead of a wrapper around it. And deprecate is_metabox_hidden().